### PR TITLE
handle nil last-updated times by setting :default to epoch

### LINKED ISSUE #21455: Doing a Slack setup on a newly started instance will not get channels until instance has been restarted
**Describe the bug**
Doing a Slack setup on a newly started instance will not get channels until instance has been restarted.
Regression in master since #18774 (PR #20570).

**To Reproduce**
1. Startup instance and setup Slack
2. Go to dashboard will show an empty channel list when trying to create a new Slack subscription

<details><summary>Full stacktrace after step 1</summary>

```
2022-04-04 19:22:53,861 ERROR middleware.log :: PUT /api/slack/settings 500 482.4 ms (13 DB calls) 
{:via
 [{:type java.lang.NullPointerException, :at [org.threeten.extra.PeriodDuration between "PeriodDuration.java" 317]}],
 :trace
 [[org.threeten.extra.PeriodDuration between "PeriodDuration.java" 317]
  [metabase.util.date_2$period_duration invokeStatic "date_2.clj" 408]
  [metabase.util.date_2$period_duration invoke "date_2.clj" 385]
  [metabase.util.date_2$older_than_QMARK_ invokeStatic "date_2.clj" 460]
  [metabase.util.date_2$older_than_QMARK_ invoke "date_2.clj" 452]
  [metabase.integrations.slack$needs_refresh_QMARK_ invokeStatic "slack.clj" 215]
  [metabase.integrations.slack$needs_refresh_QMARK_ invoke "slack.clj" 214]
  [metabase.integrations.slack$refresh_channels_and_usernames_when_needed_BANG_ invokeStatic "slack.clj" 233]
  [metabase.integrations.slack$refresh_channels_and_usernames_when_needed_BANG_ invoke "slack.clj" 230]
  [metabase.api.slack$fn__76818$fn__76821 invoke "slack.clj" 36]
  [metabase.api.slack$fn__76818 invokeStatic "slack.clj" 23]
  [metabase.api.slack$fn__76818 invoke "slack.clj" 13]
  [compojure.core$wrap_response$fn__27923 invoke "core.clj" 160]
  [compojure.core$wrap_route_middleware$fn__27907 invoke "core.clj" 132]
  [compojure.core$wrap_route_info$fn__27912 invoke "core.clj" 139]
  [compojure.core$wrap_route_matches$fn__27916 invoke "core.clj" 151]
  [compojure.core$routes$fn__27935$f__27936 invoke "core.clj" 198]
  [compojure.core$routes$fn__27935 invoke "core.clj" 200]
  [metabase.server.middleware.auth$enforce_authentication$fn__66219 invoke "auth.clj" 14]
  [compojure.core$routes$fn__27935$f__27936 invoke "core.clj" 198]
  [compojure.core$routes$fn__27935 invoke "core.clj" 200]
  [compojure.core$make_context$handler__27963 invoke "core.clj" 289]
  [compojure.core$make_context$fn__27967 invoke "core.clj" 299]
  [compojure.core$routes$fn__27935$f__27936 invoke "core.clj" 198]
  [compojure.core$routes$fn__27935$f__27936$respond_SINGLEQUOTE___27937 invoke "core.clj" 197]
  [compojure.core$make_context$fn__27967 invoke "core.clj" 300]
  [compojure.core$routes$fn__27935$f__27936 invoke "core.clj" 198]
...
```

</details>

**Information about your Metabase Installation:**
master

### PR DESCRIPTION
**Pull request by escherize**

### CODE REVIEW

File: src/metabase/integrations/slack.clj [hunk 1]
```diff
@@ -42,11 +42,15 @@
   :visibility :internal
   :type :json)
 
+(def ^:private zoned-time-epoch (t/zoned-date-time 1970 1 1 0))
```
Comment (qnkhuat) [inline src/metabase/integrations/slack.clj:45]:
```suggestion
(def ^:private zoned-1970 (t/zoned-date-time 1970 1 1 0))
```
wdyt about this name? I think it's more expressive. 
Also if you want it to be a bit poetic: `zoned-beginning-of-time`

File: src/metabase/integrations/slack.clj [hunk 2]
```diff
@@ -42,11 +42,15 @@
...
+
 (defsetting slack-channels-and-usernames-last-updated
   "The updated-at time for the [[slack-cached-channels-and-usernames]] setting."
   :visibility :internal
   :cache?     false
-  :type       :timestamp)
+  :type       :timestamp
+  ;; if it has not been set, use a very early last-updated time:
```
Comment (qnkhuat) [inline src/metabase/integrations/slack.clj:52]: I think this comment is not very helpful and IMO the field kinda explains itself so I'm fine if we don't have this comment here.

File: test/metabase/integrations/slack_test.clj [hunk 3]
```diff
@@ -236,3 +236,7 @@
               (is (= false (slack/valid-token? "abc")))
               (is (= {} (mt/summarize-multipart-email #"Your Slack connection stopped working.")))
               (is (slack/slack-token-valid?)))))))))
+
+(deftest slack-cache-updated-at-nil
```
Comment (qnkhuat) [inline test/metabase/integrations/slack_test.clj:240]: wdyt if we put this test in the `update-slack-settings-test` under `metabase.api.slack-test`?
I think it'll have more context there other than just checking if it has default value here.

**Review by qnkhuat — APPROVED** [review]
I'm glad this is fixed, I encountered this a couple of days ago when trying to setup Slack but I forgot to log it 😅 .
I had some comments but those are not blocked.
Also it'd be very helpful if you put more context in the description for whom not familar with this issue.

File: src/metabase/integrations/slack.clj (hunk 1 above)
Comment (noahmoss) [reply src/metabase/integrations/slack.clj:45]: @qnkhuat counterpoint, the term [_Unix epoch_](https://en.wikipedia.org/wiki/Unix_time) specifically refers to 0 UTC on Jan 1 1970, so I think it's probably appropriate to use _epoch_ here as well

Comment (qnkhuat) [reply src/metabase/integrations/slack.clj:45]: ahh, agreed. TIL about that term. Thanks

File: test/metabase/integrations/slack_test.clj (hunk 3 above)
Comment (escherize) [reply test/metabase/integrations/slack_test.clj:240]: Good idea! I'll add it there as well.

Comment (nemanjaglumac) [reply test/metabase/integrations/slack_test.clj:240]: FYI, Slack test under `metabase.api.slack-test` is failing across all Java versions in CircleCI:
https://app.circleci.com/pipelines/github/metabase/metabase/32972/workflows/e625bb8c-e231-4eca-a0ff-acfd219b236a.

I reran failed jobs to see if that's going to fix it. But either way, this is something that @escherize should take a look at.

**Pull request merged**